### ISSUE TITLE
URLchecker has no concept of branch

### ISSUE THREAD

**Issue by vsoch** [urlstechie/urlchecker-action#20]
Currently, we are checking out a git repo directly:

https://github.com/urlstechie/URLs-checker/blob/master/check.py#L15

However, this has no concept of branch - we would want to check out the branch for whatever PR is being run. We noticed this when running a PR for the devel branch, but seeing the current content of master.

I can open a PR to fix this up, maybe we can get it integrated with the next release that also includes #19 ?

**Comment by vsoch** [comment]
One other small detail - the git path is used for the clone, for a PR from a different user this would not work. If we trust the url check to be run for someone else's repo, then it's safe to change this to default to whatever the github repository of the PR is set to be.

**Comment by vsoch** [comment]
hey @SuperKogito - I'm curious, why do you require a clone and then delete of a repository anyway? This could be easily handled by just using the `actions/checkout@v2` repository, and then you can run the url checker directly without needing to specify a custom branch or url. It (I think) will typically be the case that we want to run it on the repository in question. As another option, we could make the git_path url paramter optional, and if not defined, it skips over the entire clone and delete at the end and just runs on the present repository that is already cloned. Let me know what you think - if we do substantial changes it would be easier to do after #19 is merged.

**Comment by SuperKogito** [comment]
I agree to the points raised here. I am still not sure about the alternative solution you suggested tbh, I still think it has some drawbacks and users will only test the repository where the action is defined (which -to your point- is supposed to the case). 

The reason why I used cloning and removing folders in the first place, was just because it was simple and easy to implement for me, so I did not spend much time looking for alternatives. I remember I looked into direct access to files but that involved some GitHub API commands, which I was not familiar with so I settled for this solution.

This is also related to a different idea that I have been having lately and I would be interested to hear your opinion on it, how about publishing a spin-off of this action as a python package? do you think it is worth it? is it useful? or should we just put all our focus on the action? 
 
I think supporting different branches is good to have but I would like to push this back to the next release. I think after testing the whole thing now, I will let out a new release if you are okay with it.

**Comment by vsoch** [comment]
Let me think about the python package - it could be useful and would require a small refactor of the client (I could help with this).

Let’s focus on the action for now. I’d like to propose allowing to specify an optional git path and branch, and if they aren’t specified, assuming that we run in the present working directory. I can open a PR for this with an updated example tomorrow! 

For the python package, let’s discuss after that. I’ve made a ton of packages and could probably whip this out very quickly, but best that we figure out the already existing action first.

**Comment by SuperKogito** [comment]
Do we need to change something for the action to run on a certain branch? 
If I am correct, this seems to be already covered by Git. Please refer to [link1](https://help.github.com/en/actions/reference/events-that-trigger-workflows) and [link2](https://stackoverflow.com/questions/58139406/only-run-job-on-specific-branch-with-github-actions).

**Comment by vsoch** [comment]
The workflow just needs the actions checkout https://github.com/actions/checkout to be the first step, and then the correct repo and branch is present in the container (mounted as a volume) in the container (and there is an environment variable they recommend using https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables). I’ll put full examples and docs for this in my work tomorrow.

**Comment by SuperKogito** [comment]
That will be a great help. I think it would good to have the example(s) in the README, that way the users won't have to look for it.

**Comment by vsoch** [comment]
Will do!

**Comment by SuperKogito** [comment]
Btw are you okay for a new release once I finish some checks and fixes?

**Comment by SuperKogito** [comment]
@vsoch  version 0.1.6 is out, feel free to run some tests and report on the branch feature and the overall functioning of the action.

**Comment by vsoch** [comment]
Woot! Yes I’ll get this tested asap!

**Comment by vsoch** [comment]
it worked beautifully for buildtest! Here is a screenshot if you'd like one of the retry - note that we got a timeout (443) and then the retry was successful (green)

![image](https://user-images.githubusercontent.com/814322/76683336-f49cc200-65c8-11ea-9117-0068e1247fa7.png)

Also note this was done via a checkout on a custom branch of a PR. I'll copy paste the config here in case others want to follow suite:

```
name: Check URLs

on: [push, pull_request]

jobs:
  urlcheck:
    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v2
    - name: URLs-checker
      uses: urlstechie/URLs-checker@0.1.6
      with:
        # check this subfolder only
        subfolder: docs

        # A comma-separated list of file types to cover in the URL checks
        file_types: .md,.py,.rst

        # Choose whether to include file with no URLs in the prints.
        print_all: false

        # How many times to retry a failed request (defaults to 1)
        retry_count: 3

        # Cannot check private GitHub settings
        white_listed_patterns: https://github.com/HPC-buildtest/buildtest-framework/settings
```
Going to test for usrse shortly, but closing issue here since it's fixed and released. Thank you @SuperKogito !
